src/main/java/edu/ie3/util/io/FileIOUtils.java

```diff
@@ -141,6 +157,495 @@ public static BufferedWriter getBufferedWriter(
+          } catch (FileException e) {
+            throw new CompletionException(e.getMessage(), e.getCause());
+          }
+        });
+  }
+
+  /**
+   * Compresses a single file and returns a {@link Future} with the result.
+   *
+   * @param fileName path of the file that should be compressed
+   * @param outputFileName path of the output file where the compressed content will be stored (path
+   *     should be for a file and filename should have a ".gz" extension)
+   * @return a Future containing a boolean which is either true on success or false otherwise
+   * @throws CompletionException If the input file name is null or the input path is not of a valid
+   *     file or if the archive already exists or unable to create the output file
+   */
+  public static CompletableFuture<Boolean> compressFile(
+      final Path fileName, final Path outputFileName) throws CompletionException {
+    return CompletableFuture.supplyAsync(
+        () -> {
+          try {
+            File validatedInputFile = validateInputFileName(fileName);
+            File validatedOutputFile =
+                validateOutputFileName(
+                    validatedInputFile.toString(), outputFileName.toString(), false);
+
+            try (GZIPOutputStream out =
+                new GZIPOutputStream(new FileOutputStream(validatedOutputFile))) {
+              try (FileInputStream in = new FileInputStream(validatedInputFile)) {
+                byte[] buffer = new byte[1024];
+                int len;
+                while ((len = in.read(buffer)) != -1) {
+                  out.write(buffer, 0, len);
+                }
+              }
+            } catch (IOException e) {
+              logger.error(
+                  "Unable to write from file '{}' to file '{}'",
+                  validatedInputFile,
+                  validatedOutputFile,
+                  e);
+              return false;
+            }
+            return true;
+
+          } catch (FileException e) {
+            throw new CompletionException(e.getMessage(), e.getCause());
+          }
+        });
+  }
+
+  /**
+   * Compress all the files present in the provided directory and returns a boolean value denoting
+   * success or failure.
+   *
+   * @param dirName path of the directory that should be compressed
+   * @param validatedOutputFile validated output file (target archive file)
+   * @return a boolean which is either true on success or false otherwise
+   * @throws FileException If unable to write to the output stream or there is some error in the
+   *     output filepath
+   */
+  private static boolean compressDir(final Path dirName, File validatedOutputFile)
+      throws FileException {
+    Path validatedOutputFileName;
+
+    try {
+      validatedOutputFileName = validatedOutputFile.toPath();
+    } catch (InvalidPathException e) {
+      throw new FileException("Invalid output file path '" + validatedOutputFile + "'.", e);
+    }
```

Comment:
Is this needed, if you validate the `validatedOutputFile` as well? At least please merge the validations a one place.

Reply:
Yes, this is not needed. I will remove the validation.